src/engine/Server.cpp

```diff
@@ -841,6 +842,55 @@ Awaitable<void> Server::processQuery(
   co_return;
 }
 
+// ____________________________________________________________________________
+Awaitable<void> Server::processUpdate(
+    const ad_utility::url_parser::ParamValueMap& params, const string& update,
+    ad_utility::Timer& requestTimer,
+    const ad_utility::httpUtils::HttpRequest auto& request, auto&& send,
+    TimeLimit timeLimit) {
+  auto queryHub = queryHub_.lock();
+  AD_CORRECTNESS_CHECK(queryHub);
+  ad_utility::websocket::MessageSender messageSender{
+      getQueryId(request, update), *queryHub};
+
```

Comment:
These 4 lines can probably also be a small helper function, because they are duplicated in the `processQuery` method.
You can then also add a small test for that.